### ISSUE TITLE
bug(ivy): ngComponentDef.h is not a function

### ISSUE THREAD

**Issue by lacolaco** [angular/angular#22922]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

There is 2 components, `AppComponent` and `GreeetingComponent`

```
import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  template: '<app-greeting name="World"></app-greeting>'
})
export class AppComponent {}
```

```
import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-greeting',
  template: `Hello {{name}}!`
})
export class GreetingComponent {
  @Input() name: string;

  foo() {}
}
```

Then, `ngc` compiles these with `enableIvy` and it succeeds.
Also, `ng build --aot` works well.

But at runtime, an error occurs.

```
_greeting_greeting_component__WEBPACK_IMPORTED_MODULE_1__.GreetingComponent.ngComponentDef.h is not a function
```

Here is the code. `i1.GreetingComponent.ngComponentDef.h(1, 0);` is an invalid call. `ComponentDef` type doesn't have `h` method. 

```
export class AppComponent {}
/** @nocollapse */
AppComponent.ctorParameters = () => [];
AppComponent.ngComponentDef = i0.ɵdefineComponent({
  type: AppComponent,
  tag: 'app-root',
  factory: function AppComponent_Factory() {
    return new AppComponent();
  },
  template: function AppComponent_Template(ctx, cm) {
    if (cm) {
      i0.ɵE(0, i1.GreetingComponent, _c0);
      i0.ɵe();
    }
    i0.ɵp(0, 'name', i0.ɵb('World'));
    i1.GreetingComponent.ngComponentDef.h(1, 0);
    i0.ɵr(1, 0);
  }
});
```

I don't have any idea to solve it. What's the expected compilation output? 

## Minimal reproduction of the problem with instructions

1. Clone https://github.com/lacolaco/ngv6-ivy
2. `yarn install`
3. `yarn build`
4 Open `dist` directory in browser with any local server

## Environment

<pre><code>
Angular CLI: 6.0.0-beta.6
Node: 8.9.3
OS: darwin x64
Angular: 6.0.0-rc.0-7cd8e8d
... common, compiler, core, platform-browser
... platform-browser-dynamic

@angular/cli: 6.0.0-beta.6
@angular/compiler-cli: 6.0.0-rc.0-f461f43
@angular/language-service: 6.0.0-rc.0-f461f43
@angular-devkit/architect: 0.0.7
@angular-devkit/build-optimizer: 0.4.6
@angular-devkit/build-webpack: 0.0.7
@angular-devkit/core: 0.4.6
@angular-devkit/schematics: 0.4.6
@ngtools/json-schema: 1.2.0
@ngtools/webpack: 6.0.0-beta.6
@schematics/angular: 0.4.6
@schematics/package-update: 0.4.6
typescript: 2.7.2
webpack: 4.1.1
</code></pre>

**Comment by trotyl** [comment]
Should be expected, the `h` property has been renamed to `hostBindings` and not used in compiled template any more (https://github.com/angular/angular/pull/22833), there're also `n` -> `factory` renaming, `ɵinject` -> `ɵdirectiveInject` renaming etc, all of them are pending corresponding compiler implementations.

**Comment by lacolaco** [comment]
@trotyl Thank you for information! 
So it should be solved already in rc.0. why I'm hitting it yet?

**Comment by trotyl** [comment]
I strongly believe one of the following will happen:

+ Ivy won't be feature complete in 6.0.0 final;
+ The 6.0.0 final will be rescheduled to make Ivy work.

Just guessing, also want to get more information from Angular team.

**Comment by pkozlowski-opensource** [comment]
@lacolaco please don't open issue for the new ivy compiler & runtime just yet, this is still work in progress and many things are working, than changing the next day etc. There is high turnaround in the code base and we don't want to track all the things through GH issues just yet. 

I know that everyone is eager to get their hands on ivy and believe me that we are excited as well and want to get it out as soon as possible. But it will take a bit more time till things are finalised and polished. 

@trotyl we are not delaying 6.0.0 final to wait for ivy, 6.0.0 will be released with what we've got for ivy at this point. Hard to say now if it will be completed 90% or 100% or some other number. Estimating software deliverables _is_ hard...

**Comment by marclaval** [comment]
By the way, this particular issue should be fixed as part of #22921
